File: 2023-2/Clases/clase_17_Herencia/circulo.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   circulo.cpp
 * Author: ASUS
 * 
 * Created on October 17, 2023, 9:14 AM
 */

#include "circulo.hpp"
#include <iostream>
#include <iomanip>
using namespace std;
#define PI 3.141592
Circulo::Circulo() {
}

Circulo::Circulo(const Circulo& orig) {
}

Circulo::~Circulo() {
}

void Circulo::SetRadio(double radio){
    this->radio = radio;
}
double Circulo::GetRadio(){
    return this->radio;
}
double Circulo::Area(){
    return PI*radio*radio;
}
double Circulo::Circunferencia(){
    return 2*PI*radio;
}
void Circulo::MostrarResultados(){
    cout.precision(2);
    cout<<fixed;
    cout<<"Radio:      "<<setw(10)<<radio<<endl;
    cout<<"Area:      "<<setw(10)<<Area()<<endl;
    cout<<"Circunferencia:      "<<setw(10)<<Circunferencia()<<endl;
}

File: 2025-1/Clase30/QuintoProyecto/src/quintoproyecto/Circulo.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package quintoproyecto;

/**
 *
 * @author hecto
 */
public class Circulo {
    private String id;
    private double radio;
    private final double PI= 3.131592;
    
    public Circulo(){
        
    }
    
    public Circulo(String id, double radio){
        this.id = id;
        this.radio = radio;
    }
    
    public Circulo(String id){
        this.id = id;   
    }                                                               

    /**
     * @return the id
     */
    public String getId() {
        return id;
    }

    /**
     * @param id the id to set
     */
    public void setId(String id) {
        this.id = id;
    }

    /**
     * @return the radio
     */
    public double getRadio() {
        return radio;
    }

    /**
     * @param radio the radio to set
     */
    public void setRadio(double radio) {
        this.radio = radio;
    }
    
    public double area(){
        return PI*radio*radio;
    }
    
    public double perimetro(){
        return 2*PI*radio;
    }
}
